R-package/R/xgboost.R

```diff
@@ -949,6 +949,244 @@ xgboost <- function(
+#' - `"response"`: will output model predictions on the scale of the response variable (e.g.
+#'  probabilities of belonging to the last class in the case of binary classification). Result will
+#'  be either a numeric vector with length matching to rows in `newdata`, or a numeric matrix with
+#'  shape `[nrows(newdata), nscores]` (for objectives that produce more than one score per
+#'  observation such as multi-class classification or multi-quantile regression).
+#' - `"raw"`: will output the unprocessed boosting scores (e.g. log-odds in the case of objective
+#'   `binary:logistic`). Same output shape and type as for `"response"`.
+#' - `"class"`: will output the class with the highest predicted probability, returned as a `factor`
+#'   (only applicable to classification objectives) with length matching to rows in `newdata`.
+#' - `"leaf"`: will output the terminal node indices of each observation across each tree, as an
+#'   integer matrix of shape `[nrows(newdata), ntrees * nscores * n_parallel_trees]`.
+#'
+#'   If the model produces more than one score per observation, columns will be arranged by putting
+#'   all the leafs for one of the scores first, followed by all of the leafs for the next score, and
+#'   so on (e.g. output for a multi-class classifier will have columns ordered as 'class1:tree1',
+#'   'class1:tree2', ..., 'class2:tree1', 'class2:tree2', etc.). If there are more than 1 parallel
+#'   trees (e.g. in a random forest), then that will be the last element in the sequence.
+#'
+#'   Only applicable to tree-based boosters (not `gblinear`).
+#' - `"contrib"`: will produce per-feature contribution estimates towards the model score for a
+#'   given observation, based on SHAP values. The contribution values are on the scale of
+#'   untransformed margin (e.g., for binary classification, the values are log-odds deviations from
+#'   the baseline).
+#'
+#'   Output will be a numeric matrix with shape `[nrows, nfeatures+1]`, with the intercept being the
+#'   last feature, or `[nrows, nscores, nfeatures+1]` if the model produces more than one score per
+#'   observation.
+#' - `"approxcontrib"`: same as `"contrib"`, but uses a fast approximation for the computation of
+#'   contributions.
+#' - `"interaction"`: similar to `"contrib"`, but computing SHAP values of contributions of
+#'   interaction of each pair of features. Note that this operation might be rather expensive in
+#'   terms of compute and memory.
+#'
+#'   Since it quadratically depends on the number of features, it is recommended to perform
+#'   selection of the most important features first.
+#'
+#'   Output will be a numeric matrix of shape `[nrows, nfeatures+1, nfeatures+1]`, or shape
+#'   `[nrows, nscores, nfeatures+1, nfeatures+1]` (for objectives that produce more than one score
+#'   per observation).
+#' - `"approxinteraction"`: similar to `"interaction"`, but uses a fast approximation for the
```

Comment:
I think the `approxcontrib` method was added at the time to demonstrate that the traditional approach has bias and SHAP doesn't, not as an optimization. If I am not mistaken, it's mostly for academic reasons to have a comparison.
We are considering whether it should continue to exist, and we definitely don't want to "advertise" it as a faster alternative.

Reply:
Removed the approx options.